app/views/includes/scripts.njk

```diff
@@ -57,3 +57,19 @@
   });
 </script>
 <script src="{{ js_path }}"></script>
+<script>
+{# If our JS bundle fails to load we need to undo Apple Pay check #}
+function ready(fn) {
+  if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading"){
+    fn();
+  } else {
+    document.addEventListener('DOMContentLoaded', fn);
```

Comment:
Not sure we should be mixing `DOMContentLoaded` and `document.readyState`. If my recollection of https://html.spec.whatwg.org/multipage/parsing.html#the-end is correct, `DOMContentLoaded` fires before `document.readyState` is set to `"completed"`.

Could we have a check to see if `document.readyState` is "interactive" or "complete" and, if it isn’t, set an listener to listen for the `readystatechange` change event and do the check again?

Reply:
This makes it work in more browsers — http://youmightnotneedjquery.com/#ready

Reply:
Hmm, still not sure we should be mixing `DOMContentLoaded` and `readyState` like this. I would have thought that browsers that support `document.readyState` would also support the `readystatechange` event.

Would it be possible to not have the second inline `script` element at all and instead have the first inline `script` element do its thing then set an event listener to listen for the `DOMContentLoaded` event, defining the function that should run right there? That would eliminate the need to check the `readyState` at all because it would also run when `DOMContentLoaded` fires.

Reply:
I’ve updated it all now, could you rereview?